File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.py
import math

def solution(n):
    ans = 0
    
    def prime(num):
        for i in range(2, int(math.sqrt(num)) + 1):
            if num % i == 0:
                return 0
        return 1
    
    for i in range(2, n + 1):
        ans += prime(i)
    
    return ans

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    const ans = new Set()
    
    const isPrime = (num) => {
        if(num < 2) return false
        for(let i=2;i<=Math.sqrt(num);i++){
            if(num % i === 0) return false
        }
        return true
    }
    
    const DFS = (fix, arr) => {
        if(arr.length === 0) return
        
        arr.forEach((item, index) => {
            if(isPrime(+(fix + item))) {
                ans.add(+(fix + item))
            }
            const temp = [...arr]
            temp.splice(index, 1)
            DFS(fix + item, temp)
        })
    }
    
    DFS("", [...numbers])
    
    return ans.size
}